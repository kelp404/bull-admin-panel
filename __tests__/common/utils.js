const utils = require('../../lib/common/utils');

afterEach(() => jest.restoreAllMocks());

test('renders base html', () => {
  const template = utils.getBaseTemplate();
  const config = {
    assetsPath: '/bull/assets',
    basePath: '/bull'
  };

  expect(template({config})).toMatchSnapshot();
});
