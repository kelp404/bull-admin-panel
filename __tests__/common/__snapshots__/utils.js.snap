// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\"/>
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no\\"/>
  <title>Bull Admin Panel</title>

  <link type=\\"text/css\\" rel=\\"stylesheet\\" href=\\"/bull/assets/web.css\\"/>
  <script type=\\"text/javascript\\">
    function base64DecodeUnicode(text) {
      return decodeURIComponent(atob(text).split('').map(function (c) {
        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
      }).join(''));
    }
    var config = JSON.parse(base64DecodeUnicode('eyJhc3NldHNQYXRoIjoiL2J1bGwvYXNzZXRzIiwiYmFzZVBhdGgiOiIvYnVsbCJ9'));
  </script>
</head>
<body>
<div id=\\"root\\">
  <div class=\\"text-center text-muted py-5 main-content\\">
    <div class=\\"spinner-border\\">
      <span class=\\"sr-only\\">Loading...</span>
    </div>
  </div>
</div>
<script type=\\"text/javascript\\" src=\\"/bull/assets/web.js\\"></script>
</body>
</html>
"
`;
